aten/src/ATen/NumericUtils.h

```diff
@@ -50,6 +50,39 @@ inline C10_HOST_DEVICE bool _isnan(at::BFloat16 val) {
   return at::_isnan(float(val));
 }
 
+// isfinite
+template <typename T,
+          typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
+inline C10_HOST_DEVICE bool _isfinite(T val) {
+  return true;
+}
+template <typename T,
+          typename std::enable_if<std::is_floating_point<T>::value, int>::type = 0>
+inline C10_HOST_DEVICE bool _isfinite(T val) {
+#if defined(__CUDACC__) || defined(__HIPCC__)
+  return ::isfinite(val);
+#else
+  return std::isfinite(val);
+#endif
+}
+
+template <typename T,
+          typename std::enable_if<c10::is_complex<T>::value, int>::type = 0>
+inline bool _isfinite(T val) {
+  return std::isfinite(val.real()) && std::isfinite(val.imag());
+}
+
+template <typename T,
+         typename std::enable_if<std::is_same<T, at::Half>::value, int>::type = 0>
+inline C10_HOST_DEVICE bool _isfinite(T val) {
+  return at::_isfinite(float(val));
+}
+
+
+inline C10_HOST_DEVICE bool _isfinite(at::BFloat16 val) {
+  return at::_isfinite(float(val));
```

Comment:
Should this be implemented as a conversion -> isfinite call or using std::numeric_limits, which PyTorch implements for bfloat16 (and float16 above, I think). 

If it's going to use a cast it should be a C++ style cast: `static_cast<float>(...)`

Reply:
I think using the `isfinite` is more consistent. Will use static_cast.